src/pages/NgramViewer.js

```diff
@@ -84,52 +92,59 @@ function NgramViewer() {
   //   return arr.map((x) => `${(x * 100).toString()} %`);
   // }
 
-  const getNgram = async () => {
+  const getNgram = async (word) => {
     await axios
       .get(
-        `https://intense-dusk-96795.herokuapp.com/https://books.google.com/ngrams/json?content=${searchInput}&year_start=1959&year_end=2019&corpus=26&smoothing=3`
+        `https://intense-dusk-96795.herokuapp.com/https://books.google.com/ngrams/json?content=${word}&year_start=1959&year_end=2019&corpus=26&smoothing=3`
       )
       .then((response) => {
         setFrequencyData(response.data[0].timeseries);
         setDatesLabel(1959, 2019);
       })
-      .catch((err) => setError(err.message, 'No results found. Please try a different input!'));
+      .catch((err) => {
+        setError('No results found. Please try a different input!');
+        console.log(err);
+      });
   };
 
   const handleInput = (e) => {
-    // resetGraph();
     setCurrentInput(e.target.value);
   };
 
-  // const resetGraph = () => {
-  //   setSearchInput('');
-  //   setFrequencyData([]);
-  //   setDatesLabel([]);
-  // };
+  const resetGraph = () => {
+    setFrequencyData([]);
+    setDatesLabel([]);
+  };
 
-  const handleSelection = (e) => {
-    // resetGraph();
-    const selectionString = [...e.target.value];
-    setCurrentSelection(selectionString.join('').replaceAll(',', ''));
+  const handleSelection = (value) => {
+    setCurrentSelection(value);
+    console.log(value);
     console.log(currentSelection);
   };
 
   const resetForm = (e) => {
     e.preventDefault();
     setCurrentSelection('');
     setCurrentInput('');
+    setSearchInput('');
   };
 
-  const handleSearchSubmit = (e) => {
+  const handleSearchSubmit = (e, userInput) => {
     e.preventDefault();
-    if (currentInput !== '') {
-      if (isValidInput(currentInput)) {
-        setSearchInput(currentInput);
-        getNgram();
+    if (currentInput !== '' && currentSelection === '') {
```

Comment:
how your brain works ill never understand this if statement hurts my brain hahahaha

Reply:
this line compares if the a dropdown value is selected but no input in the user input field (i think this might be redundant since the field is disabled when drop down is in use but just in case hehe)